polyskel.py

```diff
@@ -446,7 +446,27 @@ def show(self):
 		for item in self.__data:
 			print(item)
 
-
+def merge_sources(skeleton):
+    """ Multiple sources may share same location 
+    :param skeleton:
+    :return:
+    """
+    sources = {}
+    to_remove = []
+    for i, p in enumerate(skeleton):
+        if p.source in sources:
```

Comment:
On Python v3.8.2 this gives me "TypeError: unhashable type: Point2". Which is reasonable. What configuration worked for you?

Also, since Point2 works with floats, I wonder if this hashing approach is safe in general? What with float comparison and things.